### PR TITLE
Reduce integration test overhead

### PR DESCRIPTION
**Pull request by bohde**
In profiling integration tests, I found a couple places where per-test overhead could be reduced:

* Avoiding disk IO by synchronizing instead of deleting & copying test Git repository data. This saves ~100ms per test on my machine
* When flushing queues in `PrintCurrentTest`, invoke `FlushWithContext` in a parallel.

### CODE REVIEW

**Review by lunny — APPROVED** [review]

**Comment by wxiaoguang** [comment]
> When flushing queues in `PrintCurrentTest`, invoke `FlushWithContext` in a parallel.

We could make it even faster (without any waiting time). Could I edit your PR to implement my idea?

**Comment by bohde** [comment]
> We could make it even faster (without any waiting time). Could I edit your PR to implement my idea?

I'd be fine with this!

File: models/unittest/fscopy.go [hunk 1]
```diff
@@ -32,67 +30,75 @@ func Copy(src, dest string) error {
...
-	if len(filters) > 0 {
-		filter = filters[0]
+	for _, file := range files {
+		destFilePath := filepath.Join(destPath, file)
+		if _, err = os.Stat(filepath.Join(srcPath, file)); err != nil {
+			if os.IsNotExist(err) {
+				// TODO: why delete? it should happen because the file list is just queried above, why not exist?
+				if err := os.RemoveAll(destFilePath); err != nil {
+					return err
+				}
```
Comment (wxiaoguang) [inline models/unittest/fscopy.go]: ~~I have a question here, why "IsNotExist" would happen?~~

Hmm .... I see

File: modules/testlogger/testlogger.go [hunk 2]
```diff
@@ -118,7 +118,7 @@ func PrintCurrentTest(t testing.TB, skip ...int) func() {
 				_, _ = fmt.Fprintf(os.Stdout, "+++ %s ... still flushing after %v ...\n", t.Name(), SlowFlush)
 			}
 		})
-		if err := queue.GetManager().FlushAll(context.Background(), time.Minute); err != nil {
+		if err := queue.GetManager().FlushAll(context.Background(), -1); err != nil {
```
Comment (wxiaoguang) [inline modules/testlogger/testlogger.go:121]: * Old code: very long time
* New code (errgroup + go): about 40ms in total
* New code (this one: discard all): less than 1ms

**Review by wxiaoguang — APPROVED** [review]

**Comment by wxiaoguang** [comment]
Done from my side.

**Comment by bohde** [comment]
I think these changes look good!

**Comment by wxiaoguang** [comment]
Need a following up PR:  Improve testing and try to fix MySQL hanging #32515

**Pull request merged**